### PR TITLE
bip327: minor fixes

### PR DESCRIPTION
**Pull request by siv2r**
I made some minor corrections, which I believe are beneficial. Please let me know if any of them are incorrect or need modification.

Changes:
1. An error test vector doesn’t specify the `InvalidContributionError` type
2. In *DeterministicSign*, use `GetXonlyPubkey` instead of `GetPubkey` (doesn’t exist)
3. The `key_agg_and_tweak` fn doesn’t specify the return type
4. In `partial_sig_verify_internal`, the pubkey arg type should be `PlainPk` rather than `bytes`
5. Remove unnecessary `enumerate()` calls
    - I suspect this was initially used in debugging to find the failing test vector, and then it was forgotten.
6. In `test_sign_verify`, add an additional `assert` statement that checks if two pubnonces add up to the infinity aggnonce.

cc @jonasnick @real-or-random

### CODE REVIEW

**Comment by siv2r** [comment]
Additionally, I removed the following `P is None` check since `cpoint` never returns `None`.
https://github.com/bitcoin/bips/blob/812907c2b00b92ee31e2b638622a4fe14a428aee/bip-0327/reference.py#L442-L444

Also, we should return `False` when `cpoint` throws an exception when parsing `P,` `R_s1`, and `R_s2`. Currently, it `partial_sig_verify_internal` simply crashes.

File: bip-0327/reference.py [hunk 1]
```diff
@@ -598,7 +596,9 @@ def test_sign_verify_vectors() -> None:
 
     aggnonces = fromhex_all(test_data["aggnonces"])
     # The aggregate of the first three elements of pnonce is at index 0
-    assert(aggnonces[0] == nonce_agg([pnonce[0], pnonce[1], pnonce[2]]))
+    assert (aggnonces[0] == nonce_agg([pnonce[0], pnonce[1], pnonce[2]]))
+    # The aggregate of the first and fourth elements of pnonce is at index 1
```
Comment (stratospher) [inline bip-0327/reference.py:600]: b6d7143: nit: maybe mention context about them adding to infinity like in the PR writeup? (it wasn't immediately obvious to me.)

**Review by stratospher — COMMENTED** [review]
ACK 2fc3ef4.

File: bip-0327/vectors/sig_agg_vectors.json [hunk 2]
```diff
@@ -143,7 +143,8 @@
             ],
             "error": {
                 "type": "invalid_contribution",
-                "signer": 1
+                "signer": 1,
+                "contrib": "psig"
```
[the remark this answers is not in the document]
Comment (siv2r) [reply bip-0327/vectors/sig_agg_vectors.json:147]: Ahh yes. I missed it. Thanks!

File: bip-0327/reference.py (hunk 1 above)
Comment (siv2r) [reply bip-0327/reference.py]: Fixed

**Review by jonasnick — COMMENTED** [review]
Thanks @siv2r! Here's a commit that update the BIP to 1.0.2: https://github.com/jonasnick/bips/commits/minor-fixes-jn/. Feel free to cherry-pick.

> We should probably return False when cpoint throws an exception when parsing P, R_s1, and R_s2. Currently, the partial_sig_verify_internal simply crashes.

That would also be possible, but it wouldn't have an effect on users of the API. Right now, `partial_sig_verify_internal` raises a ValueError that cannot be triggered by any of the public functions and that is tested by test vectors.

**Comment by siv2r** [comment]
> Feel free to cherry-pick.

Thanks, I cherry-picked it! Just changed my name.
> That would also be possible, but it wouldn't have an effect on users of the API.

Hmm, we can avoid making this change. If we proceed, we would also need to place the `get_session_key_agg_coeff` function call in a try-except block. Too many try-except blocks would make the code less readable.

I initially wanted this change because `partial_sig_verify_internal` was called directly (not through `verify`) in [one of the tests](https://github.com/bitcoin/bips/blob/master/bip-0327/reference.py#L748). However, it should be fine since it's a test code.

**Comment by siv2r** [comment]
Please don't merge it yet. The changelog says `PartialSigVerify` vector was updated. It should instead be `PartialSigAgg`. I'll fix it later.

**Comment by siv2r** [comment]
Fixed the changelog to say `PartialSigAgg`

**Comment by murchandamus** [comment]
@jonasnick: Could you take another look?

**Review by jonasnick — APPROVED** [review]
ACK 26bb1d8ea3e2f0f7e02e1ec37a4b70fbc0781f85

**Pull request merged**